app/forms/form.rb

```diff
@@ -9,8 +9,22 @@ class Form
 
   delegate :persisted?, to: :claim
 
-  def self.model_name
-    Claim.model_name
+  class << self
+    def model_name
+      Claim.model_name
+    end
+
+    def slug_name
+      name.demodulize.delete_suffix("Form").underscore.dasherize
+    end
+
+    # Returns form classes outside of any journey-specific namespace
+    def all_shared_forms
+      ObjectSpace
+        .each_object(Class)
+        .select { |klass| klass < self }
+        .reject { |klass| klass.to_s.start_with?("Journeys::") }
+    end
```

Comment:
How does this perform? I think these loops are going to do quite a few iterations. Maybe best to do this sort of thing in an initialiser so it doesn't have to be done on each request?

Reply:
Good shout, it was actually extremely slow! I've fixed it by using class variables.